### PR TITLE
grass.pygrass: remove unused arg in ctypes.CFUNCTYPE

### PR DESCRIPTION
**Pull request by mshukuno**
Please see the comment [here](https://github.com/OSGeo/grass/pull/3928#discussion_r1685058517).

Initially, the PR was about fixing "ambiguous variable name 'l', but in this case 'l's were also unused so deletion was the initial fix. However, it appears Windows has a "ctypes" issue.

I would like to revert them as they were, but rename "l"s to avoid the E741 (Checks for the use of the characters 'l', 'O', or 'I' as variable names) issue. Since I couldn't figure out what "l" parameters are so I renamed it "l_what" for now. Let me know if you find appropriate variable name. Here is [pygrass.raster package documentation](https://grass.osgeo.org/grass83/manuals/libpython/pygrass.raster.html#module-pygrass.raster.rowio).

However, the flake8 problem in F841 (unused-variable) will be addressed next, and noqa:F841 will be applied to them when I work on F841 problem.

I propose raising a separate issue if this needs to be investigated further since I don't have experience with C and "ctypes".

What do you think?

### CODE REVIEW

File: python/grass/pygrass/raster/rowio.py [hunk 1]
```diff
@@ -18,17 +18,17 @@
 )
```
Comment (petrasovaa) [inline python/grass/pygrass/raster/rowio.py:18]: Seems like removing the last `ctypes.c_int` together with removing the `l_what` parameter could work, not tested though.

Comment (nilason) [reply python/grass/pygrass/raster/rowio.py:18]: That should be the solution! The first argument to [ctypes.CFUNCTYPE](https://docs.python.org/3/library/ctypes.html#ctypes.CFUNCTYPE) is the `restype`, followed by the argument types of the function (`getmaprow_[CELL|FCELL|DCELL](int, pointer, int)`).

@mshukuno Please try the following change Anna suggested:

```diff
diff --git a/python/grass/pygrass/raster/rowio.py b/python/grass/pygrass/raster/rowio.py
index 377843642a..ed43c9a845 100644
--- a/python/grass/pygrass/raster/rowio.py
+++ b/python/grass/pygrass/raster/rowio.py
@@ -13,9 +13,7 @@ from grass.pygrass.errors import GrassError
 from grass.pygrass.raster.raster_type import TYPE as RTYPE
 
 
-CMPFUNC = ctypes.CFUNCTYPE(
-    ctypes.c_int, ctypes.c_int, ctypes.c_void_p, ctypes.c_int, ctypes.c_int
-)
+CMPFUNC = ctypes.CFUNCTYPE(ctypes.c_int, ctypes.c_int, ctypes.c_void_p, ctypes.c_int)
 
 
 def getmaprow_CELL(fd, buf, row):
```

**Comment by nilason** [comment]
The reason for this to fail on the win runner only, could be the fact that that is the only one with Python 3.12.

**Comment by nilason** [comment]
@echoix I have enabled auto-merge, if you agree with me this is good to go, just approve.

**Comment by echoix** [comment]
I would only need to search the logs, this one, and just before, to see it fixed. I can't search on my phone, only scroll. So I'm checking soon ;) (unless someone finds the lines before).

**Comment by echoix** [comment]
A run in main before this PR (after fixing trim of GEOS WKT output): https://github.com/OSGeo/grass/actions/runs/10217611481/job/28271801017#step:11:1483
This PR: https://github.com/OSGeo/grass/actions/runs/10218185138/job/28273664754?pr=4113#step:11:11141

**Review by echoix — APPROVED** [review]

**Pull request merged**